toolbox/src/test/scala/org/finos/toolbox/json/JsonUtilTest.scala

```diff
@@ -0,0 +1,46 @@
+package org.finos.toolbox.json
+
+import org.finos.toolbox.json.JsonUtil.{fromJson, toRawJson}
+import org.scalatest.featurespec.AnyFeatureSpec
+import org.scalatest.matchers.should.Matchers
+
+class JsonUtilTest extends AnyFeatureSpec with Matchers {
+
+  Feature("toRawJson") {
+    Scenario("can convert an instance of a case class to json string") {
+      val dto = TestDto(name = "orange", value = 13.7, quantity = Some(5))
+
+      val result = toRawJson(dto)
+
+      result shouldEqual """{"name":"orange","value":13.7,"quantity":5}"""
+    }
+  }
+
+  Feature("fromJson") {
+    Scenario("can create an instance of a case class from json string") {
+      val jsonAsString = """{"name": "apple", "value": 20.5, "quantity": 7}"""
+
+      val result = fromJson[TestDto](jsonAsString)
+
+      result shouldEqual TestDto(name = "apple", value = 20.5, quantity = Some(7))
+    }
+
+    Scenario("falls back to `None` when an `Option` field is missing") {
+      val jsonAsString = """{"name": "orange", "value": 20.5}"""
+
+      val result = fromJson[TestDto](jsonAsString)
+
+      result shouldEqual TestDto(name = "orange", value = 20.5, quantity = None)
+    }
+
+    Scenario("falls back to `None` when an `Option` field is null") {
```

Comment:
Yeah I did explore that as well, the results for different types were different. For instance, `String -> null`, `Double -> 0.0`, `Int -> 0`. Probably because of how `Object` works in Java. I did a bit of research to have it resolved in a straightforward way, couldn't find anything concrete but I believe there are some not so straightforward workarounds this so that all values fall back to `null` regardless of the type.

However, I finally settled with `Option` as I think it won't be very `Scala`ish to allow `null`s. So it's probably better practice that fields that are optional or can be null should have an `Option` type in `Scala`.